crates/cxx-qt-gen/src/generator/cpp/property/getter.rs

```diff
@@ -3,27 +3,38 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use crate::generator::{cpp::fragment::CppFragment, naming::property::QPropertyNames};
+use crate::generator::{
+    cpp::fragment::CppFragment,
+    naming::property::{NameState, QPropertyNames},
+};
 use indoc::formatdoc;
 
-pub fn generate(idents: &QPropertyNames, qobject_ident: &str, return_cxx_ty: &str) -> CppFragment {
-    CppFragment::Pair {
-        header: format!(
-            "{return_cxx_ty} const& {ident_getter}() const;",
-            ident_getter = idents.getter.cxx_unqualified()
-        ),
-        source: formatdoc!(
-            r#"
-            {return_cxx_ty} const&
-            {qobject_ident}::{ident_getter}() const
-            {{
-                const ::rust::cxxqt1::MaybeLockGuard<{qobject_ident}> guard(*this);
-                return {ident_getter_wrapper}();
-            }}
-            "#,
-            ident_getter = idents.getter.cxx_unqualified(),
-            ident_getter_wrapper = idents.getter_wrapper.cxx_unqualified(),
-        ),
+pub fn generate(
+    idents: &QPropertyNames,
+    qobject_ident: &str,
+    return_cxx_ty: &str,
+) -> Option<CppFragment> {
+    if let NameState::Auto(name) = &idents.getter {
+        Some(CppFragment::Pair {
+            header: format!(
+                "{return_cxx_ty} const& {ident_getter}() const;",
+                ident_getter = name.cxx_unqualified()
+            ),
+            source: formatdoc!(
+                r#"
+                    {return_cxx_ty} const&
+                    {qobject_ident}::{ident_getter}() const
+                    {{
+                        const ::rust::cxxqt1::MaybeLockGuard<{qobject_ident}> guard(*this);
+                        return {ident_getter_wrapper}();
+                    }}
+                    "#,
+                ident_getter = name.cxx_unqualified(),
+                ident_getter_wrapper = idents.getter_wrapper.cxx_unqualified(),
+            ),
+        })
+    } else {
+        None
     }
 }
 
```

Comment:
shouldn't the `generate_wrapper` below also use `if let NameState::Auto(name) = &idents.getter_wrapper {` for completeness ?

Reply:
ah i see `getter_wrapper` is a `Name` not `NameState` or `Option<NameState>`, so probably ignore this then

Reply:
changed